src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -196,6 +198,160 @@ public static partial class AdvApi32
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool GetTokenInformation(
+            SafeObjectHandle TokenHandle,
+            TOKEN_INFORMATION_CLASS TokenInformationClass,
+            void* TokenInformation,
+            uint TokenInformationLength,
+            out uint ReturnLength);
+
+        /// <summary>
+        ///     The QueryServiceObjectSecurity function retrieves a copy of the security descriptor associated with a service
+        ///     object. You can also use the GetNamedSecurityInfo function to retrieve a security descriptor.
+        /// </summary>
+        /// <param name="hService">
+        ///     A handle to the service control manager or the service. Handles to the service control manager
+        ///     are returned by the <see cref="OpenSCManager" /> function, and handles to a service are returned by either the
+        ///     <see cref="OpenService" /> or <see cref="CreateService" /> function. The handle must have the READ_CONTROL access
+        ///     right.
+        /// </param>
+        /// <param name="dwSecurityInformation">
+        ///     A set of bit flags that indicate the type of security information to retrieve. This
+        ///     parameter can be a combination of the <see cref="SECURITY_INFORMATION" /> flags, with the exception that this
+        ///     function does not support the <see cref="SECURITY_INFORMATION.LABEL_SECURITY_INFORMATION" /> value.
+        /// </param>
+        /// <param name="lpSecurityDescriptor">
+        ///     A pointer to a buffer that receives a copy of the security descriptor of the
+        ///     specified service object. The calling process must have the appropriate access to view the specified aspects of the
+        ///     security descriptor of the object. The SECURITY_DESCRIPTOR structure is returned in self-relative format.
+        /// </param>
+        /// <param name="cbBufSize">
+        ///     The size of the buffer pointed to by the <paramref name="lpSecurityDescriptor" /> parameter, in
+        ///     bytes. The largest size allowed is 8 kilobytes.
+        /// </param>
+        /// <param name="pcbBytesNeeded">
+        ///     A pointer to a variable that receives the number of bytes needed to return the requested
+        ///     security descriptor information, if the function fails.
+        /// </param>
+        /// <returns>
+        ///     If the function succeeds, the return value is a nonzero value.
+        ///     <para>
+        ///         If the function fails, the return value is zero. To get extended error information, call
+        ///         <see cref="GetLastError" />.
+        ///     </para>
+        /// </returns>
+        [DllImport(nameof(AdvApi32), SetLastError = true)]
+        public static extern bool QueryServiceObjectSecurity(
+            SafeServiceHandle hService,
+            SECURITY_INFORMATION dwSecurityInformation,
+            byte[] lpSecurityDescriptor,
```

Comment:
Why is this parameter typed as a byte array instead of something like `SECURITY_DESCRIPTOR`?